### ISSUE TITLE
Some widgets don't show up on mac startup

### ISSUE THREAD

**Issue by nicolopadovan** [exelban/stats#1360]
**Describe the bug**

**Details:**
 - Device: Macbook Pro 2021, M1 Max
 - macOS: Ventura 13.2.1
 - Application version: 2.8.12

Most widgets, on mac startup, are not visibile - even though the widgets are active in the settings.
Toggling them off and back on solves the issue, but this must be done every time I turn the computer on.
The problem may be related to the fact that I am using an ultra wide monitor, and maybe the app is calculating the available horizontal space based on the macos built in retina screen (which is 16", but the space is limited due to the notch).

**Comment by exelban** [comment]
What modules are not enabled on startup?
And what exactly do you need to toggle in the settings, the module toggle, or the widget itself?

**Comment by nicolopadovan** [comment]
> What modules are not enabled on startup? And what exactly do you need to toggle in the settings, the module toggle, or the widget itself?

![Screenshot 2023-03-18 at 13 53 51](https://user-images.githubusercontent.com/89971888/226107234-79f6d98a-7375-4f8b-a5d8-ba90d86c6814.png)

I need to toggle this switch off and on again, for each widget (RAM, CPU, GPU...)
While I don't think it's related to which toggle, but how far it is from the right side of the screen, the modules I have been experiencing the problem with are:
- CPU
- GPU
- RAM
- DISK

This doesn't appear to be happening to:
- Sensors
- Network
- Battery

Network and battery are the farthest right modules I have, but sensors is actually the farthest left (which suggests that either the sensors module is not being affected or that the position is not related to the problem.
![Screenshot 2023-03-18 at 13 56 42](https://user-images.githubusercontent.com/89971888/226107683-caac9cb6-adeb-41c4-a16e-74e195b29f54.png)
Screenshot of my menu bar for completeness

**Comment by exelban** [comment]
Please disable all modules except the CPU. Leave one active widget for the CPU module. Restart the application and check if the widget will be enabled.

**Comment by nicolopadovan** [comment]
> Please disable all modules except the CPU. Leave one active widget for the CPU module. Restart the application and check if the widget will be enabled.

I think I mismatched between widget and module in my previous messages. 
The problem is not with "control center" widgets (or better: it may be, but I don't use the control center at all), but only in the menu bar icons.
Would the solution you provided be effective nonetheless?

**Comment by exelban** [comment]
widget - it's an element in the menu bar.

**Comment by nicolopadovan** [comment]
> Please disable all modules except the CPU. Leave one active widget for the CPU module. Restart the application and check if the widget will be enabled.

I disabled all modules except the CPU - restarting the app, the widget is in fact enabled.

**Comment by kaspy003** [comment]
Yes, i have same problem.. every time when i start my mac studio (m1 max) i need to open app and toggle CPU checkbox to enable it ;(

**Comment by exelban** [comment]
@kaspy003 please see my previous comment)

**Comment by nicolopadovan** [comment]
> > Please disable all modules except the CPU. Leave one active widget for the CPU module. Restart the application and check if the widget will be enabled.
> 
> I disabled all modules except the CPU - restarting the app, the widget is in fact enabled.

Tried this fix, but it didn't work, and it appears that the problem is actually related to how far right each widget is: I tried changing the order, and the first on the right (in this case SSD) does show up, the others need to be toggled again.
Again, this doesn't effect sensors.

![Screenshot 2023-03-23 at 10 02 49](https://user-images.githubusercontent.com/89971888/227153934-4b3c2253-b9d3-49c9-a808-96f7b1f29bbb.png)

**Comment by exelban** [comment]
please remove all elements from the menu bar and try again. Looks like you are trying to show more elements in the menu bar than it can fit.

**Comment by nicolopadovan** [comment]
> please remove all elements from the menu bar and try again. Looks like you are trying to show more elements in the menu bar than it can fit.

It's true, to some extent: I am trying to show more elements than the menu bar on the Retina Display can fit; however, my other monitor does have the space to show all of the items.
Probably, the widgets are being added as soon as the mac is logged in, and while it's still refreshing the available monitors.
A possible fix would be to reload the widgets after this process has finished.

**Comment by exelban** [comment]
Another. When you have enough space on your main screen do the widgets shows normally on start?

**Comment by nicolopadovan** [comment]
> When you have enough space on your main screen do the widgets shows normally on start?

It seems so, yes.
